### ISSUE TITLE
trampoline throws an error 

### ISSUE THREAD

**Issue by dpsutton** [technomancy/leiningen#2628]
- [x] Updating to using the latest released version of Leiningen (`lein upgrade`).
- [x] Moving your `~/.lein/profiles.clj` (if present) out of the way. This contains third-party dependencies and plugins that can cause problems inside Leiningen.
- [x] Updating any old versions of plugins in your `project.clj`, especially if the problem is with a plugin not working. Old versions of plugins like nREPL and CIDER (as well as others) can cause problems with newer versions of Leiningen.
- [ ] (If you are using Java 9 or newer), updating your dependencies to their most recent versions. Recent JDK's have introduced changes which can break some Clojure libraries.

**Describe the bug**
`lein trampoline repl` does not work on 2.9.1

**To Reproduce**
```
[dan@fedora queue]$ rm -rf ~/.m2/repository/nrepl/
[dan@fedora queue]$ lein --version
Leiningen 2.9.1 on Java 1.8.0_232 OpenJDK 64-Bit Server VM
[dan@fedora queue]$ lein trampoline repl
Retrieving nrepl/nrepl/0.6.0/nrepl-0.6.0.pom from clojars
Retrieving nrepl/nrepl/0.4.5/nrepl-0.4.5.pom from clojars
Retrieving nrepl/bencode/1.0.0/bencode-1.0.0.pom from clojars
Retrieving nrepl/drawbridge/0.1.0/drawbridge-0.1.0.pom from clojars
Retrieving nrepl/nrepl/0.4.0/nrepl-0.4.0.pom from clojars
Retrieving nrepl/nrepl/0.6.0/nrepl-0.6.0.jar from clojars
Retrieving nrepl/drawbridge/0.1.0/drawbridge-0.1.0.jar from clojars
ClassCastException java.lang.String cannot be cast to java.lang.Number
	nrepl.server/start-server/addr--1161 (server.clj:120)
	nrepl.server/start-server (server.clj:128)
	nrepl.server/start-server (server.clj:96)
	reply.eval-modes.nrepl/get-connection (nrepl.clj:158)
	reply.eval-modes.nrepl/get-connection (nrepl.clj:156)
	reply.eval-modes.nrepl/main (nrepl.clj:202)
	reply.eval-modes.nrepl/main (nrepl.clj:200)
	reply.main/launch-nrepl/fn--2327 (main.clj:79)
	clojure.core/with-redefs-fn (core.clj:7434)
	clojure.core/with-redefs-fn (core.clj:7418)
	reply.main/launch-nrepl (main.clj:78)
	reply.main/launch-nrepl (main.clj:72)
Bye for now!

```

**Link to sample project**
bug happens in any project i've tried it in. Example is `manifold`.

**Environment**
- Leiningen Version: Leiningen 2.9.1 on Java 1.8.0_232 OpenJDK 64-Bit Server VM
- Leiningen installation method: been a while but i think the install script and since then `lein upgrade <version>`
- JDK Version: openjdk version "1.8.0_232"
- OS: fedora 31

**Additional context**

I installed a local version of `nrepl` and it seems that `nrepl.server/start-server` is being called with `"0"` as a port rather than an integer or nil (which will default to 0 in nrepl). Not sure if this is a regression in nrepl not handling an integer port or reply changing how it calls start server.

This behavior is not observed with lein 2.8.3.

I discovered this trying to start a socket repl for chlorine following their documented startup command:
`JVM_OPTS='-Dclojure.server.myrepl={:port,5555,:accept,clojure.core.server/repl}' lein trampoline repl`.

I found the minimal repro was `lein trampoline run` in any lein project.

**Comment by dpsutton** [comment]
It's possible this is already fixed in reply.

https://github.com/trptcolin/reply/pull/198

Lein 2.9.1 is using reply 4.3 which lacks the patch 

```
(nrepl.server/start-server :port port)
```
vs the fix in the PR:
```
(nrepl.server/start-server
  :port (when port
          (-> port str Integer/parseInt)))
```

So perhaps this issue goes away in the next release?

**Comment by glts** [comment]
Yeah, this was fixed with #2615.

**Comment by dpsutton** [comment]
thanks @glts I'll close. I wonder if its possible to get a release soon?

**Comment by glts** [comment]
Yeah, a new release would be nice. But we’ll need a new Pomegranate release (including an outstanding pull request there) to properly fix the CVE at #2551. Perhaps some other things too …
